File: defaultcode/cpp/command/command.cpp
#include <iostream>

using namespace std;

// Command Interface
class Command
{
public:
	virtual void execute();
};
 
// Receiver Class
class $referenceclass 
{
public:
	
}; 

// Invoker 
class $invokerclass {

public:
	void setCommand(Command *cmd) {
		this->cmd = cmd;
	}

	void buttonPressed() {
		cmd->execute();
	} 

private:
	Command *cmd;

};

// Command class
class $classname : public Command {

    public:

        $classname($referenceclass  *rclass){
            this->rclass=rclass;
        }

        void execute(){}

    private:
        $referenceclass  *rclass;
};
 
 

File: defaultcode/python/command/command.py
from abc import ABC, abstractmethod

#command interface
class Command(ABC):

    @abstractmethod
    def execute(self):
        pass

class $referenceclass:
    pass

#invoker class
class $invokerclass:

    def __init__(self, command):
        self.cmd = command

    def ButtonPressed(self):
        self.cmd.execute()

#command class
class $classname(Command):
   
    def __init__(self, refclass):
        self.ref = refclass

    def execute(self):
       pass

       